pre-commit-config.yaml

```diff
@@ -24,63 +26,52 @@ repos:
     - id: trailing-whitespace
     - id: mixed-line-ending
       args: ['--fix=lf']
-    - id: flake8
-      additional_dependencies: [
-      'flake8-blind-except',
-      'flake8-commas',
-      'flake8-comprehensions',
-      'flake8-deprecated',
-      'flake8-docstrings',
-      'flake8-meiqia',
-      'flake8-mutable',
-      'flake8-pep3101',
-      'flake8-quotes',
-      'flake8-string-format',
-      'flake8-tidy-imports',
-      # These are good but are broken with current pydocstyle version
-      # 'flake8-print',
-      # 'flake8-debugger',
-      # 'flake8-todo',
-      ]
 
-- repo: git@github.com:humitos/mirrors-autoflake.git
+- repo: https://github.com/humitos/mirrors-autoflake.git
   rev: v1.1
   hooks:
     - id: autoflake
       args: ['--in-place', '--remove-all-unused-imports', '--remove-unused-variable']
 
-- repo: git@github.com:pre-commit/mirrors-yapf.git
-  rev: v0.26.0
-  hooks:
-    - id: yapf
-      exclude: 'migrations|tests'
-      args: ['--style=.style.yapf', '--parallel', '--in-place']
-
-# When calling `pre-commit autoupdate`, this repo needs to be commented because it fails
-- repo: git@github.com:FalconSocial/pre-commit-python-sorter.git
-  rev: b57843b0b874df1d16eb0bef00b868792cb245c2  # higher than 1.0.4 (latest release)
-  hooks:
-    - id: python-import-sorter
-      args: ['--silent-overwrite']
-      exclude: 'wsgi.py'
+# TODO: re-enable it and use `--range` to select the modified lines
+# - repo: https://github.com/myint/docformatter.git
+#   rev: v1.4
+#   hooks:
+#     - id: docformatter
+#       args: ['--in-place', '--wrap-summaries=80', '--wrap-descriptions=80', '--pre-summary-newline']
 
-- repo: git@github.com:myint/docformatter.git
-  rev: v1.1
-  hooks:
-    - id: docformatter
-      args: ['--in-place', '--wrap-summaries=80', '--wrap-descriptions=80', '--pre-summary-newline']
+# TODO: re-enable when we can make it behave in the same way than in CircleCI
+# - repo: https://github.com/PyCQA/prospector
+#   rev: 1.3.1
+#   hooks:
+#     - id: prospector
+#       args: ['--profile=/home/humitos/rtfd/code/readthedocs.org/prospector.yml']
+#       additional_dependencies: [
+#       # TODO: use an environment variable to decide this PATH here?
+#       '-r/home/humitos/rtfd/code/readthedocs.org/requirements/lint.txt',
+#       ]
+#       # NOTE: copied from prospector YAML configuration because pre-commit does
+#       # not load them from there and runs prospector over these files
+#       exclude: |
+#             (?x)^(
+#                 docs/.*|
+#                 readthedocs/settings/.*|
+#                 common/.*|
+#                 readthedocs/rtd_tests/.*|
+#                 readthedocs/.*tests.*/.*|
+#                 .*/migrations/.*|
+#                 .*conftest.py
+#             )$
```

Comment:
Or, in the interest in reducing the number of moving pieces, prospector might be another optional tool. We used this in the past to glue several tools together, but black might remove the need for this entirely anyways.

Reply:
In that case, should we remove `prospector` from CircleCI as well? My idea was to try to emulate running all the checks that are run in CircleCI locally via pre-commit. Otherwise, we will be passing the checks locally but they may fail when running in CircleCI.

Reply:
Good question, I don't have strong opinions. You've spent more time configuring linting, so if you think it's unnecessary, I'd at least be +1 on testing what linting looks like locally and at CI without prospector. I imagine it's safe to remove though, prospector predates pre-commit and yapf. The rest of backend team might have opinions here though

Reply:
Hrm, I don't think we can remove `prospector` from our linting passes because it detects modules required not being imported in the file, for example. However, we could probably remove some of the style checks that prospector does to speed up this particular check since black will be in charge of this.